imap/index.c

```diff
@@ -5647,10 +5550,42 @@ static int extract_attachment(const char *type, const char *subtype,
 
 EXPORTED void index_text_extractor_init(struct protstream *clientin)
 {
+    const char *exturl =
+         config_getstring(IMAPOPT_SEARCH_ATTACHMENT_EXTRACTOR_URL);
+    if (!exturl) return;
+
     syslog(LOG_DEBUG, "extractor_init(%p)", clientin);
 
+    char scheme[6], server[100], path[256], *p;
+    unsigned https, port;
+
+    /* Parse URL (cheesy parser without having to use libxml2) */
+    int n = sscanf(exturl, "%5[^:]://%99[^/]%255[^\n]",
+                   scheme, server, path);
+    if (n != 3 ||
+        strncmp(lcase(scheme), "http", 4) || (scheme[4] && scheme[4] != 's')) {
+        syslog(LOG_ERR,
+               "extract_attachment: unexpected non-HTTP URL %s", exturl);
+        return;
+    }
+
+    /* Normalize URL parts */
+    https = (scheme[4] == 's');
+    if (*(p = path + strlen(path) - 1) == '/') *p = '\0';
+    if ((p = strrchr(server, ':'))) {
+        *p++ = '\0';
+        port = atoi(p);
+    }
+    else port = https ? 443 : 80;
+
+    /* Build servername, port, and options */
+    struct buf buf = BUF_INITIALIZER;
+    buf_printf(&buf, "%s:%u%s/noauth", server, port, https ? "/tls" : "");
+
     index_text_extractor = xzmalloc(sizeof(struct extractor_ctx));
     index_text_extractor->clientin = clientin;
+    index_text_extractor->path = xstrdup(path);
+    index_text_extractor->hostname = buf_release(&buf);
```

Comment:
So all the parsing code got moved into here, and this is the only place exturl is referenced.  We just cache those two things.

Hmm, I realise I probably didn't create a cleanup for those - incoming commit in a sec.